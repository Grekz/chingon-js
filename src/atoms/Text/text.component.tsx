import styled from '@emotion/styled'
import * as React from 'react'

interface ITextProps {
  children?: React.ReactNode | string
  type?: TextType
  color?: string
}

type TextType = 'h1' | 'h2' | 'h3' | 'span'

const TextWrapper = (props: ITextProps) => {
  const { children, color, type } = props
  switch (type) {
    case 'h1':
      return <H1Wrapper color={color}>{children}</H1Wrapper>
    case 'h2':
      return <H2Wrapper color={color}>{children}</H2Wrapper>
    case 'h3':
      return <H3Wrapper color={color}>{children}</H3Wrapper>
    case 'span':
      return <SpanWrapper color={color}>{children}</SpanWrapper>
    default:
      return <SpanWrapper color={color}>{children}</SpanWrapper>
  }
}

const Text = (props: ITextProps) => {
  const { type, color, children } = props
  return TextWrapper({ color, children, type })
}

Text.defaultProps = {
  children: 'Sample text',
  color: '#000',
  type: 'span',
}

const SpanWrapper = styled.span`
  color: ${props => props.color};
`

const H1Wrapper = styled.h1`
  color: ${props => props.color};
`

const H2Wrapper = styled.h2`
  color: ${props => props.color};
`

const H3Wrapper = styled.h3`
  color: ${props => props.color};
`

export default Text
