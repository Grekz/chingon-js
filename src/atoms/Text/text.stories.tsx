import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import Text from '.'

storiesOf('Atoms|Text', module).add('As Default', () => (
  <>
    <Text color="red" type="h1">
      Hello
    </Text>
    <Text color="magenta" type="h2">
      Hello
    </Text>
    <Text type="h3">Hello</Text>
    <Text type="span">Hello</Text>
  </>
))
