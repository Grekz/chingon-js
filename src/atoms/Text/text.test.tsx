import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import Text from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<Text /> rendering', () => {
  it('should render Text h1', () => {
    const wrapper = shallow(<Text type="h1" />)
    expect(wrapper).toHaveLength(1)
  })
  it('should render Text h2', () => {
    const wrapper = shallow(<Text type="h2" />)
    expect(wrapper).toHaveLength(1)
  })
  it('should render Text h3', () => {
    const wrapper = shallow(<Text type="h3" />)
    expect(wrapper).toHaveLength(1)
  })
  it('should render Text span', () => {
    const wrapper = shallow(<Text type="span" />)
    expect(wrapper).toHaveLength(1)
  })
  it('should match the snapshot for Text h1', () => {
    const wrapper = shallow(<Text type="h1" />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
