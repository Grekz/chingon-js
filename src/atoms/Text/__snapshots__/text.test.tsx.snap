// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> rendering should match the snapshot for Text h1 1`] = `
<Styled(h1)
  color="#000"
>
  Sample text
</Styled(h1)>
`;
